lib/ema.js

```diff
@@ -17,26 +19,70 @@ class EMA extends Indicator {
     })
 
     this._a = 2 / (period + 1)
+    this._buffer = []
   }
 
   static unserialize (args = []) {
     return new EMA(args)
   }
 
+  _calculateEMA (val, previousEma) {
+    return new BigN(this._a).multipliedBy(val).plus(
+      new BigN(1).minus(this._a).multipliedBy(previousEma)
+    ).toNumber()
+  }
+
+  reset () {
+    super.reset()
+    this._buffer = []
+  }
+
+  v () {
+    if (this.l() === 0) {
+      return null
+    }
+    return super.v()
+  }
+
   update (v) {
-    if (this.l() < 2) {
+    if (this._seedPeriod === 1) {
       return super.update(v)
     }
 
-    return super.update((this._a * v) + ((1 - this._a) * this.prev()))
+    if (_isEmpty(this._buffer)) {
+      this._buffer.push(v)
```

Comment:
memleak?

Reply:
not a memleak.. buffer always contains the values which was used to calculate first ema